### DISCUSSION TITLE
[Question] sync hugegraph data daily with rocksdb, how to handle lots of g/logs

### DISCUSSION THREAD

**Discussion by xxyp** [apache/incubator-hugegraph#2267] [Q&A]
### Problem Type (问题类型)

None

### Before submit

- [X] 我已经确认现有的 [Issues](https://github.com/apache/hugegraph/issues) 与 [FAQ](https://hugegraph.apache.org/docs/guides/faq/) 中没有相同 / 重复问题 (I have confirmed and searched that there are no similar problems in the historical issue and documents)

### Environment (环境信息)

- Server Version: 0.11.0 
- Backend: 本地RocksDB 

### Your Question (问题描述)

定时任务往hugegraph里同步数据，使用本地rocksdb存储，在g文件夹了产生了大量日志，要怎么设置才能不让日志占满磁盘呢

**Comment by imbajin** [comment]
if u have a lot of rocksdb logs(and u dont't need them), try to just clean it with crontab?

the old log will be auto truncated by default I remember 🦙

**Comment by xxyp** [comment]
定时任务是10分钟同步一次。这个日志的最长保留时间是多少天呀，因为同步的频率太高了，没有及时清理完？
用crontab清也可以，就是麻烦一点

**Comment by imbajin** [comment]
> 定时任务是 10 分钟同步一次。这个日志的最长保留时间是多少天呀，因为同步的频率太高了，没有及时清理完？ 用 crontab 清也可以，就是麻烦一点

forever if u don't delete it? (by default)

**Comment by xxyp** [comment]
那现在除了定时清理之外，没有其他的办法可以设置了，是吗？

**Reply by imbajin** [reply]
准确的说是没有太简单的好办法, 你可以去调整一下 rocksdb 的自动设置日志的 api, 这样也可以通过 Java-RocksDB-API 控制它的日志生成策略, 只不过可能需要新增一部分代码

**Reply by xxyp** [reply]
想问一下，g m s这三个文件夹下面的000063.log 和LOG.old.1654601528979230这些格式的日志删掉，对服务有影响吗？

**Reply by imbajin** [reply]
一般这些都是 rocksdb 的生成的相关 data/meta/wal 文件为主, 类似你可以搜下它相关[文档](https://zhuanlan.zhihu.com/p/344903822)或者 GPT 询问(当然不一定精准). (看你个人需要选择)

<img width="633" alt="image" src="https://github.com/apache/incubator-hugegraph/assets/17706099/e6e9b9a4-a309-4c37-a02c-c688bc79592b">

**Comment by xxyp** [comment]
可以通过rocksdb.max_total_wal_size这个配置来控制保留的数量吗

**Reply by imbajin** [reply]
> 可以通过rocksdb.max_total_wal_size这个配置来控制保留的数量吗

可以搜一下 rocksdboption 类这块是否暴露了相关参数，当然也可以随时提交 pr 加上（如果miss了）
